relay-dynamic-config/src/project.rs

```diff
@@ -19,6 +21,35 @@ use crate::metrics::{
 use crate::trusted_relay::TrustedRelayConfig;
 use crate::{GRADUATED_FEATURE_FLAGS, defaults};
 
+/// Per-Category settings for retention policy.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RetentionSettings {
+    /// Standard / full fidelity retention policy in days.
+    pub standard: u16,
+    /// Downsampled retention policy in days.
+    pub downsampled: u16,
+}
+
+/// Settings for retention policy.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Retentions {
+    /// Setting for DataCategory.LOG_BYTE.
+    pub log: RetentionSettings,
+    /// Setting for DataCategory.SPAN.
+    pub span: RetentionSettings,
```

Comment:
Do we want these optional, or if retentions are defined they must all be defined? Latter would be nice, but I don't know if that's possible/feasible.